### DISCUSSION TITLE
Deprecation of Docker Image Resource

### DISCUSSION THREAD

**Discussion by kcbimonte** [concourse/concourse#9047] [General]
# BLUF (Bottom Line Up Front)
The Docker Image Resource was intended to be deprecated about five (5) years ago per this milestone [(Deprecate `docker-image` resource)](https://github.com/concourse/concourse/milestone/59) in favor of the [concourse/oci-build-task](https://github.com/concourse/oci-build-task) and [concourse/registry-image-resource](https://github.com/concourse/registry-image-resource) repositories. 

# Problem
The docker-image resource has continued to receive updates despite being slated for deprecation in favor of other resources that have various improvements such as safer execution or broader test coverage. The OCI Build Task hasn't seen updates in about nine (9) months (understanding the problems with the Concourse CI instance contributed to delays).

# Attempts
The docker-image was proposed to be deprecated five years ago and three (almost four) years ago with #6586 but it was tied to Prototypes, so it stalled. 

# Proposal
Targeting Concourse Release [v8.0.0](https://github.com/concourse/concourse/milestone/97), deprecation of the docker-image resource should be attempted in order to reduce the maintenance footprint of three resources down to two where the two resources are more focused instead of being one catch all resource.

**Comment by crsimmons** [comment]
One problem I see with this is the oci-build-task and registry-image resource don't cover all the usecases for the docker-image resource. For example, I don't see a way to import an image from a tgz and push to a registry via Concourse without using the docker resource.

**Reply by kcbimonte** [reply]
Is a .tgz a requirement or does a .tar work just as well for your use case? Because if a .tar works, this is the docs on how you would do that:
- https://github.com/concourse/registry-image-resource?tab=readme-ov-file#put-step-out-script-push-and-tag-an-image

A tangible example of this is found under this Concourse How to Page which I can confirm works:
- https://concourse-ci.org/building-and-pushing-an-image.html#the-entire-pipeline

**Reply by crsimmons** [reply]
That's a good point. We're using Concourse to import an image from a 3rd party to push to an internal registry and I'm not sure exactly what format the image is in. I'll give it a shot with the registry image resource.

**Reply by robtaylorsky** [reply]
Would tagging images be another use case not currently supported by oci-build-task ?

**Reply by taylorsilva** [reply]
Correct, oci-build task doesn't name or tag images for you. That would all happen when you push an image with the `registry-image` resource. I would consider that use-case covered by these two tools.

FYI, the `registry-image` resource has gotten some nice enhancements recently and it's able to work with untarred oci-layout images, which might be what OP's third-party image was using.
